src/snowcli/output/printing.py

```diff
@@ -23,6 +23,62 @@ def default(self, o):
         return super().default(o)
 
 
+class OutputData:
+    def __init__(self, format: Optional[OutputFormat] = None) -> None:
+        self.output: List[Union[SnowflakeCursor, str, List[Dict]]] = []
+        self.counter = 0
+        self._format = format
+
+    @property
+    def format(self):
+        if not self._format:
+            self._format = _get_format_type()
+        return self._format
+
+    def add_data(self, data: Union[str, SnowflakeCursor, List[Dict]]):
+        self.output.append(data)
+        self.counter += 1
+        return self
+
+    def get_data(self) -> Iterable:
```

Comment:
What is the type of items in this iterable?

Reply:
`Iterator[List[Dict]]`

Changed

Reply:
Changed to `def get_data(self) -> Union[str, Dict, Iterable[List[Dict]]]:`